amm-executor/src/Spectrum/Executor/EventSource/Persistence/LedgerHistory.hs

```diff
@@ -0,0 +1,155 @@
+module Spectrum.Executor.EventSource.Persistence.LedgerHistory
+  ( LedgerHistory(..)
+  , mkLedgerHistory
+  , mkRuntimeLedgerHistory
+  ) where
+
+import RIO
+  ( ByteString
+  , MonadThrow (throwM)
+  , ($>)
+  , newIORef
+  , readIORef
+  , writeIORef
+  , (<&>)
+  , isJust
+  )
+
+import qualified Data.ByteString.Lazy as LBS
+import qualified Data.Map as Map
+import Data.Aeson
+  ( ToJSON, encode, FromJSON, decode )
+
+import Control.Monad.IO.Class
+  ( MonadIO (liftIO) )
+import Control.Monad.Trans.Resource
+  ( MonadResource )
+
+import System.Logging.Hlog
+  ( Logging (..), MakeLogging (..) )
+
+import qualified Database.RocksDB as Rocks
+
+import Spectrum.Executor.Types
+  ( ConcretePoint )
+import Spectrum.Executor.EventSource.Persistence.Data.BlockLinks
+  ( BlockLinks )
+import Spectrum.Executor.EventSource.Persistence.Exception
+  ( StorageDeserializationFailed(StorageDeserializationFailed) )
+import Spectrum.Executor.EventSource.Persistence.Config
+  ( LedgerStoreConfig (..) )
+
+data LedgerHistory m = LedgerHistory
```

Comment:
Isn't it better to put this(and related) logic into separate module and later in a separate repo? We can implement our own explorer chain sync service using this codebase.